sympy/core/add.py

```diff
@@ -315,10 +316,17 @@ def flatten(cls, seq):
         # now let's construct new args:
         # [2*x**2, x**3, 7*x**4, pi, ...]
         newseq = []
+        imfinite = []
```

Comment:
cmplxfinite? (Not that it matters as such, but maybe a better name?)

And similar two lines down.

Reply:
Hmm...I'll change this back to a draft and think more about this.